scripts/globals/hunts.lua

```diff
@@ -0,0 +1,1616 @@
+    player:startEvent(1500,bitCombo,registryZone.huntMenu[1])
+
+  -- stops player from taking a hunt if a regime is active
+  elseif player:getCharVar("[regime]id") >= 1 then
+    local bitCombo = scyldBits + 0x0001 -- bit displays regime active menu
+    player:startEvent(1500,bitCombo,registryZone.huntMenu[1])
+
+  -- default menu
+  else
+    player:startEvent(1500,scyldBits,registryZone.huntMenu[1])
+  end
+end
+
+function tpz.hunts.onEventUpdate(player,csid,option)
+  local registryZone = zone[player:getZoneID()]
+  local region = registryZone[option]
+
+  player:updateEvent(0,0,registryZone.huntMenu[option])
+
+  -- handles region select
+  option = bit.band(option, 0xFF)
+  if registryZone.huntMenu[option] then
+    player:updateEvent(0,1,registryZone.huntMenu[option])
+  end
+
+  -- gets progress of current hunt (param controls the display of kills needed)
+  if option == 3 then
+    player:updateEvent(1)
+  end
+
+  -- displays hunt info (kills required,0,0,0,zoneId,huntId,scyld bounty+fee,?)
+  if region then
+    local huntPage = registryZone.hunt[region.optionId]
+    local bountyBit = bit.lshift(huntPage.bounty, 10)
+    local feeBit = huntPage.fee
+    local scyldParam = bountyBit + feeBit
+    player:updateEvent(1,0,0,0,0,region.huntId,scyldParam,1)
+  end
+end
+
+tpz.hunts.clearHuntVars = function(player)
+  player:setCharVar("[hunt]id", 0)
+  player:setCharVar("[hunt]status", 0)
+  player:setCharVar("[hunt]zone", 0)
+  player:setCharVar("[hunt]page", 0)
+end
+
+function tpz.hunts.onEventFinish(player, csid, option)
+  local scylds = player:getCurrency("scyld")
+  local registryZone = zone[player:getZoneID()]
+  local huntEntry = registryZone.hunt[option]
+  local msg = registryZone.msg
+
+-- accepting hunt
+if huntEntry then
+    player:messageSpecial(msg[1])
+    player:messageSpecial(msg[2], huntEntry.fee, scylds)
+    player:setCharVar("[hunt]status", 1)
+    player:setCharVar("[hunt]zone", player:getZoneID())
+    player:setCharVar("[hunt]id", option)
+    player:setCharVar("[hunt]page", huntEntry.pageId)
+    player:delCurrency("scyld", huntEntry.fee)
+
+  -- cancels hunt
+  elseif option == 3 then
+    player:messageSpecial(msg[3])
+    player:setCharVar("[hunt]status", 0)
+
+  -- cancels training regime and clears all vars
+  elseif option == 4 then
+    player:messageSpecial(msg[4])
+    tpz.regime.clearRegimeVars(player)
+
+  -- completes hunt
+  elseif option == 5 then
+    local huntZone = player:getCharVar("[hunt]zone")
+    local huntId = player:getCharVar("[hunt]id")
+    local scyldBounty = zone[huntZone].hunt[huntId].bounty
+
+    player:messageSpecial(msg[5])
+    player:messageSpecial(msg[6], scyldBounty, scylds)
+    player:addCurrency("scyld", scyldBounty)
+    -- give player evoliths here
+    player:setCharVar("[hunt]nextHunt", getVanaMidnight())
+    tpz.hunts.clearHuntVars(player)
+
+    -- scylds cap at 1000
+    if player:getCurrency("scyld") + bounty > 1000 then
```

Comment:
`bounty` is nil, is this supposed to be `scyldBounty`?

Reply:
Yes, I think you're right. I'll change it.